### PR TITLE
Tolerations and TopologySpreadConstraints. Fixes #1803 and #1844

### LINKED ISSUE #1803: Tolerations not supported
Hi, are there any plans to support providing tolerations in addition to nodeAffinity?

Thank you!

### PR DESCRIPTION
**Pull request by rigazilla**
Adds spec.scheduling.tolerations spec.scheduling.topologySpreadConstraints.

see #1803 #1844

### CODE REVIEW

File: test/e2e/infinispan/smoke_test.go [hunk 1]
```diff
@@ -89,6 +103,22 @@ func verifyDefaultAuthention(require *require.Assertions, ispn *v1.Infinispan) {
...
 }
 
+// Make sure no PVCs were created
+func verifyScheduling(assert *assert.Assertions, require *require.Assertions, ispn *v1.Infinispan) {
+	pod := corev1.Pod{}
+	require.NoError(testKube.Kubernetes.Client.Get(context.TODO(), types.NamespacedName{Name: ispn.Name + "-0", Namespace: tutils.Namespace}, &pod))
+	// From Infinispan CR to pods
+	areEqual := false
+	for _, v := range pod.Spec.Tolerations {
+		if reflect.DeepEqual(v, ispn.Spec.Scheduling.Tolerations[0]) {
```
Comment (ryanemerson) [inline test/e2e/infinispan/smoke_test.go:113]: We can just do `assert.EqualValues()` here on each of the tolerations.

File: test/e2e/infinispan/smoke_test.go [hunk 2]
```diff
@@ -89,6 +103,22 @@ func verifyDefaultAuthention(require *require.Assertions, ispn *v1.Infinispan) {
...
+			areEqual = true
+			break
+		}
+	}
+	assert.True(areEqual)
+	assert.True(reflect.DeepEqual(pod.Spec.TopologySpreadConstraints, ispn.Spec.Scheduling.TopologySpreadConstraints))
```
Comment (ryanemerson) [inline test/e2e/infinispan/smoke_test.go:119]: `assert.EqualValues`

File: test/e2e/infinispan/smoke_test.go [hunk 3]
```diff
@@ -89,6 +103,22 @@ func verifyDefaultAuthention(require *require.Assertions, ispn *v1.Infinispan) {
 	testAuthentication(ispn, schema, user, pass)
 }
 
+// Make sure no PVCs were created
```
Comment (ryanemerson) [inline test/e2e/infinispan/smoke_test.go:106]: I don't think this comment is valid.

File: test/e2e/infinispan/smoke_test.go (hunk 1 above)
Comment (rigazilla) [reply test/e2e/infinispan/smoke_test.go:113]: k8s adds some tolerations, so arrays are not equal. so we need to check if our toleration is in the array

**Pull request merged**